WireAnalytics/Sources/WireAnalytics/AnalyticsSession.swift

```diff
@@ -0,0 +1,43 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Countly
+
+public final class AnalyticsSession: AnalyticsSessionProtocol {
+
+    public init(appKey: String, host: URL) {
+        let config = CountlyConfig()
+        config.appKey = appKey
+        config.host = host.absoluteString
+
+        WireCountly.sharedInstance().start(with: config)
+    }
+
+    public func startSession() {
+        WireCountly.sharedInstance().beginSession()
+    }
+
+    public func endSession() {
+        WireCountly.sharedInstance().endSession()
+    }
+
+    public func trackEvent(_ event: AnalyticEvent) {
+        WireCountly.sharedInstance().recordEvent(event.rawValue)
+    }
+
+}
```

Comment:
Could we inject the Countly instead of accessing a singleton in every method?

```suggestion
public final class AnalyticsSession: AnalyticsSessionProtocol {

    let countly: WireCountly

    public init(
        countly: WireCountly,
        appKey: String,
        host: URL
    ) {
        self.countly = countly

        let config = CountlyConfig()
        config.appKey = appKey
        config.host = host.absoluteString
        countly.start(with: config)
    }

    public func startSession() {
        countly.beginSession()
    }

    public func endSession() {
        countly.endSession()
    }

    public func trackEvent(_ event: AnalyticEvent) {
        countly.recordEvent(event.rawValue)
    }
}
```

Reply:
Fixed it by making it a struct as we discussed